File: java-application-maven-archetype/src/main/resources/archetype-resources/src/test/java/MainIT.java
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package};

import static java.lang.ProcessBuilder.Redirect.INHERIT;
import static java.nio.charset.Charset.defaultCharset;
import static org.assertj.core.api.Assertions.assertThat;

import com.google.common.io.CharStreams;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.Reader;
import org.junit.Test;

public class MainIT {

  private static final String DOCKER_IMAGE = System.getProperty("docker.image");

  @Test
  public void main()
      throws Exception {

    String actual = execute("docker", "run", "--rm", DOCKER_IMAGE).trim();
    assertThat(actual).endsWith(".Main -- Hello, World!");
  }

  private static String execute(String... commandLine)
      throws IOException, InterruptedException {

    ProcessBuilder processBuilder = new ProcessBuilder(commandLine);
    processBuilder.redirectError(INHERIT);
    Process process = processBuilder.start();

    try (Reader reader = new InputStreamReader(process.getInputStream(), defaultCharset())) {
      String output = CharStreams.toString(reader);
      if (process.waitFor() != 0) {
        throw new IOException("Command failed with exit code: " + process.exitValue());
      }
      return output;
    }
  }
}

File: kotlin-application-maven-archetype/src/main/resources/archetype-resources/src/test/kotlin/MainIT.kt
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import java.io.IOException
import java.lang.ProcessBuilder.Redirect.INHERIT
import java.nio.charset.Charset.defaultCharset

private val DOCKER_IMAGE = System.getProperty("docker.image")

private fun execute(vararg commandLine: String): String {
    val processBuilder = ProcessBuilder(*commandLine)
    processBuilder.redirectError(INHERIT)
    val process = processBuilder.start()

    if (process.waitFor() != 0) {
        throw IOException("Command failed with exit code: ${process.exitValue()}")
    }
    val output = process.inputStream.bufferedReader(defaultCharset()).readText()
    return output.trim()
}

class MainIT {

    @Test
    fun main() {
        val actual = execute("docker", "run", "--rm", DOCKER_IMAGE)
        assertThat(actual).endsWith(".Main -- Hello, World!")
    }
}
